test/test-client-server.js

```diff
@@ -1157,7 +1157,61 @@ var tests = [
     },
     what: 'client auto-rejects unrequested, allows requested forwarded-tcpip'
   },
+  { run: function() {
+      var self = this,
+          what = this.what,
+          client,
+          server,
+          wasReply = false,
+          fastrejectSent = false,
+          r;
+
+      function sendAcceptLater(accept) {
+        // Only call accept after reject was called
+        if (fastrejectSent) {
+          accept();
+        } else {
+          process.nextTick(sendAcceptLater, accept);
```

Comment:
Thanks, will correct it. What do you thing about auto-rejecting?
R

On 11 December 2015 at 09:53, Brian White <notifications@github.com> wrote:

> In test/test-client-server.js
> <https://github.com/mscdex/ssh2/pull/372#discussion_r47332687>:
>
> > @@ -1157,7 +1157,61 @@ var tests = [
> >      },
> >      what: 'client auto-rejects unrequested, allows requested forwarded-tcpip'
> >    },
> > +  { run: function() {
> > +      var self = this,
> > +          what = this.what,
> > +          client,
> > +          server,
> > +          wasReply = false,
> > +          fastrejectSent = false,
> > +          r;
> > +
> > +      function sendAcceptLater(accept) {
> > +        // Only call accept after reject was called
> > +        if (fastrejectSent) {
> > +          accept();
> > +        } else {
> > +          process.nextTick(sendAcceptLater, accept);
>
> This is failing on node v0.10 and node v0.12 because process.nextTick()
> did not accept arguments to pass to the function until iojs and node v4+.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mscdex/ssh2/pull/372/files#r47332687>.
>